ProcessMaker/Traits/HasVersioning.php

```diff
@@ -153,14 +153,15 @@ public function getPublishedVersion(array $data)
         $implementation = WorkflowManager::NAYRA_PUBLISHER . get_class($this);
         $existsCustomPublisher = $implementation
             && WorkflowManager::existsServiceImplementation($implementation);
-        if ($existsCustomPublisher) {
+        if ($existsCustomPublisher && $data !== []) {
```

Comment:
Instead of skip the custom publisher and In sync with product, we should improve the message displayed to the end user. So consider to use a try{} catch {} and throw an 422 HttpException.

Reply:
hi @caleeli I've made the changes. I encountered some issues while implementing the suggestion; however, if something is still incomplete, we can have a meeting to discuss it. I'm attaching a video of the current solution.

https://github.com/ProcessMaker/processmaker/assets/90741874/cf271140-4203-4204-af6f-b0df4437dfd1